HWs/HW1/BWT/BWT/Program.cs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        var nextSymbolsIndexes = new int[text.Length];
+        for (int i = 0; i < text.Length; ++i)
+        {
+            nextSymbolsIndexes[characterFrequencies[text[i]]] = i;
+            ++characterFrequencies[text[i]];
+        }
+
+        var nextSymbol = nextSymbolsIndexes[matrixIndex];
+        var result = new StringBuilder();
+        for (int i = 0; i < text.Length; ++i)
+        {
+            result.Append(text[nextSymbol]);
+            nextSymbol = nextSymbolsIndexes[nextSymbol];
+        }
+
+        return result.ToString();
+    }
+
+    static void Main()
+    {
+        Console.WriteLine("Hi, I can encode and decode a text using the Burrows-Wheeler transform.");
+        Console.Write("Choose what you need [encode/decode]: ");
+
+        string? action = "";
+
+        while (action != "0")
+        {
+            action = Console.ReadLine();
+            if (action == null)
+            {
+                throw new ArgumentNullException("Text cannot be null.");
+            }
+            action = action.ToLower();
+
+            switch (action)
+            {
+                case "encode":
+                    Console.Write("Good! Enter your text: ");
+                    string? textForEncode = Console.ReadLine();
+                    while (textForEncode == null)
+                    {
+                        textForEncode = Console.ReadLine();
+                    }
+
+                    var resultEncode = Encode(textForEncode);
+                    Console.WriteLine($"Converted text: {resultEncode.Item1}\nLine end number: {resultEncode.Item2}");
+                    Console.Write("Don't want to see the reverse transformation? (Enter DECODE): ");
+                    break;
+
+                case "decode":
+                    Console.Write("Good! Enter your text: ");
+                    string? textForDecode = Console.ReadLine();
+                    while (textForDecode == null || textForDecode == "")
+                    {
+                        Console.WriteLine("Incorrect input!");
+                        Console.WriteLine("The text cannot be empty.");
+                        Console.Write("Try again: ");
+                        textForDecode = Console.ReadLine();
+                    }
+
+                    Console.Write("Good! Enter the number of the original row in the sorted matrix: ");
+                    int indexMatrix = -1;
+                    while (!int.TryParse(Console.ReadLine(), out indexMatrix) ||
+                           indexMatrix < 0 || indexMatrix >= textForDecode.Length)
+                    {
+                        Console.WriteLine("Incorrect input!");
+                        Console.WriteLine("The number of the original row is a non-negative number and less than the length of the text.");
+                        Console.Write("Try again: ");
+                    }
+
+                    var resultDecode = Decode(textForDecode, indexMatrix);
```

Comment:
```suggestion
                    string resultDecode = Decode(textForDecode, indexMatrix);
```
При инициализации переменной результатом вычисления метода тип стоит явно указывать.
А `var` использовать только тогда, когда тип очевиден из правой части (например, вызов конструктора).